src/client/api/rpc.c

```diff
@@ -114,8 +115,10 @@ query_cb(struct crt_proto_query_cb_info *cb_info)
 
 	if (daos_rpc_retryable_rc(cb_info->pq_rc)) {
 		rproto->ep.ep_rank = (rproto->ep.ep_rank + 1) % rproto->nr_ranks;
+		if (rproto->timeout < 30)
```

Comment:
it would be better here that instead of 30 to use the default timeout that cart was inited with. not sure if there is an api to get default timeout from cart, but should be easy to add one and use that instead of 30 and if current timeout exceeds the default to just use the default.

Reply:
nope there isnt a single/simple way, as there are multiple timeouts involved (global, per context, per rpc).